### PR TITLE
 refactor: GLTF loader using the prefab system

### PR DESCRIPTION
**Pull request by Rhuagh**
This change is [Reviewable](https://reviewable.io/reviews/amethyst/amethyst/784)

### CODE REVIEW

File: amethyst_gltf/Cargo.toml [hunk 1]
```diff
@@ -21,13 +21,14 @@ amethyst_renderer = { path = "../amethyst_renderer/", version = "0.7.0" }
...
 fnv = "1"
-gltf = "0.9"
-gltf-utils = "0.9"
+gltf = "0.11"
 gfx = "0.17"
 hibitset = { version = "0.5.0", features = ["parallel"] }
 imagefmt = "4.0"
 itertools = "0.7"
 log = "0.4"
+mikktspace = { version = "0.1", git = "https://github.com/gltf-rs/mikktspace" }
```
Comment (Xaeroxe) [inline amethyst_gltf/Cargo.toml]: What does it take to get this on crates.io?  git dependencies block releases.

File: amethyst_gltf/src/format/mod.rs [hunk 2]
```diff
@@ -60,8 +66,10 @@ impl StdError for GltfError {
         match *self {
             GltfImporterError(_) => "Gltf import error",
             InvalidSceneGltf(_) => "Gltf has no default scene, and the number of scenes is not 1",
-            PrimitiveMissingInGfx(_) => "Primitive missing in gfx",
+            //PrimitiveMissingInGfx(_) => "Primitive missing in gfx",
```
Comment (Xaeroxe) [inline amethyst_gltf/src/format/mod.rs]: Remove commented out code please

File: amethyst_gltf/src/format/mod.rs [hunk 3]
```diff
@@ -84,10 +92,12 @@ impl fmt::Display for GltfError {
             GltfImporterError(ref err) => {
                 write!(f, "{}: {}", self.description(), err.description())
             }
-            PrimitiveMissingInGfx(ref err) => write!(f, "{}: {}", self.description(), err),
+            //PrimitiveMissingInGfx(ref err) => write!(f, "{}: {}", self.description(), err),
```
Comment (Xaeroxe) [inline amethyst_gltf/src/format/mod.rs]: Same as above

File: amethyst_gltf/src/lib.rs [hunk 4]
```diff
@@ -6,141 +6,228 @@ extern crate base64;
...
+    type Result = ();
+
+    fn load_prefab(
+        &self,
+        entity: Entity,
+        system_data: &mut Self::SystemData,
+        entities: &[Entity],
+    ) -> Result<(), Error> {
+        if let Some(ref transform) = self.transform {
+            transform.load_prefab(entity, &mut system_data.0, entities)?;
```
Comment (Xaeroxe) [inline amethyst_gltf/src/lib.rs]: Tuple is getting pretty large and should be destructured into more friendly names.

File: amethyst_gltf/Cargo.toml (hunk 1 above)
Comment (Rhuagh) [reply amethyst_gltf/Cargo.toml]: It was released yesterday, just waiting for reviews before fixing.

File: amethyst_gltf/src/format/mod.rs (hunk 2 above)
Comment (Rhuagh) [reply amethyst_gltf/src/format/mod.rs]: Right, I forgot readding the logic that use that one, good catch.

Comment (Rhuagh) [reply amethyst_gltf/src/format/mod.rs]: Ah, we only support triangle lists currently, so I'll remove for now and make a note in my todo.

File: amethyst_gltf/src/format/material.rs [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+                .dropping(offset)
+                .step(step)
+                .cloned()
+                .collect();
+            TextureData::U8(image_data, metadata)
+        }
+        TextureData::Rgba(ref color, ref metadata) => {
+            TextureData::Rgba([color[offset]; 4], metadata.clone())
+        }
+        _ => unreachable!(), // We only support color and image for textures from gltf files
```
Comment (torkleyy) [inline amethyst_gltf/src/format/material.rs]: Is this truly unreachable?

File: amethyst_gltf/src/format/mesh.rs [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+                        &n,
+                        &tx,
+                        &mut set_tangent,
+                    );
+                }
+
+                tangents.iter().map(|(_, t)| [t[0], t[1], t[2]]).collect()
+            });
+
+        debug!("Loading bounding box");
```
Comment (torkleyy) [inline amethyst_gltf/src/format/mesh.rs]: Those are many debug calls, are you sure we need them all / they should log at debug?

File: amethyst_gltf/src/format/material.rs (hunk 5 above)
Comment (Rhuagh) [reply amethyst_gltf/src/format/material.rs]: Yes, GLTF only support material data with either a [f32; 4] or an image.

Comment (Rhuagh) [reply amethyst_gltf/src/format/material.rs]: And those are the only variants that are produced by the material loading.

**Comment by Rhuagh** [comment]
bors r=torkleyy

**Pull request merged**